framework/tests/aliasing/ExplicitAnnotationTest.java

```diff
@@ -0,0 +1,13 @@
+import org.checkerframework.common.aliasing.qual.Unique;
+
+@Unique class Data {
+    @SuppressWarnings("unique.leaked")
+    Data() {} // All objects of Data are now @Unique
+}
+
+class Demo {
+    void check(Data p) { // p is @Unique Data Object
+        // :: error: (unique.leaked)
+        Data y = p; // @Unique p is leaked
```

Comment:
Is the type of the left-hand side important? Can you add another method that does

`Object check2(Data p) { return p; }`

which I guess should also give a `unique.leaked` error.

Reply:
Ping.

Reply:
No the type of the left-hand side isn't important. However, the method doesn't give an error, even when Data is explicitly annotated as @Unique.

Reply:
I'm not sure how you mean this. Does `Object z = p;` not raise an error? Shouldn't it?

Reply:
```
@Unique
class Data {

    Object check2(Data p) { return p; }

    void check(Data p) {
        Object z = p;
    }
}
```
In the code given above, Object z = p; does raise an error. Hence, the left hand side isn't important. However, the check2 function (where you are returning a @Unique Data as an Object) doesn't raise an error both in the typetools/master and in my modified code, even if the Data p is explicitly annotated as @Unique Data p.

Reply:
Ok, can you expand this test case with the `Object` local variable and the expected error.
Then file an issue that shows the problem with the `return` version.